src/lib/server/models.ts

```diff
@@ -84,11 +73,65 @@ const modelConfig = z.object({
 
 const modelsRaw = z.array(modelConfig).parse(JSON5.parse(MODELS));
 
+async function getChatPromptRender(
+	m: z.infer<typeof modelConfig>
+): Promise<ReturnType<typeof compileTemplate<ChatTemplateInput>>> {
+	if (m.chatPromptTemplate) {
+		return compileTemplate<ChatTemplateInput>(m.chatPromptTemplate, m);
+	} else {
+		let tokenizer: PreTrainedTokenizer;
+
+		if (!m.tokenizer) {
+			throw new Error(
+				"No tokenizer specified and no chat prompt template specified for model " + m.name
+			);
+		}
+
+		try {
+			tokenizer = await getTokenizer(m.tokenizer);
+		} catch (e) {
+			throw Error(
+				"Failed to load tokenizer for model " +
+					m.name +
+					" consider setting chatPromptTemplate manually or making sure the model is available on the hub."
+			);
+		}
+
+		const renderTemplate = ({ messages, preprompt }: ChatTemplateInput) => {
+			let formattedMessages: { role: string; content: string }[] = messages.map((message) => ({
+				content: message.content,
+				role: message.from,
+			}));
+
+			if (preprompt) {
+				formattedMessages = [
+					{
+						role: "system",
+						content: preprompt,
+					},
+					...formattedMessages,
+				];
+			}
+
+			const output = tokenizer.apply_chat_template(formattedMessages, {
```

Comment:
yep, I tried looking into it, but because the error messages are [hardcoded in the jinja template](https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.2/blob/main/tokenizer_config.json#L42), it's not obvious to me how to determine if the template is failing bc of a lack of system prompt or for some other reason

I guess you could retry without the `"system"` message and see if it builds the prompt then? we could do it in a second PR if that sounds good to you ?